include/fire/Event.h

```diff
@@ -350,14 +398,20 @@ class Event {
       //   they are new and not from an input file
       auto& obj{objects_[full_name]};
       obj.data_ = std::make_unique<io::Data<DataType>>(io::constants::EVENT_GROUP+"/"+full_name);
-      obj.should_save_ = keep(full_name, GET_KEEP_DEFAULT);
+      obj.should_save_ = tag_it->keep();
       obj.should_load_ = true;
       obj.updated_ = false;
+
       // get this object up to the current entry
       //    loading may throw an H5 error if the shape of the data on disk
       //    cannot be loaded into the input type
       try {
-        for (std::size_t i{0}; i < i_entry_+1; i++) input_file_->load_into(*obj.data_);
+        if (not obj.should_save_) {
+          // only skip the first i_entry_ entries if this object is not being saved
+          //  the objects that are being saved are being mirrored by the input file
+          for (std::size_t i{0}; i < i_entry_; i++) input_file_->load_into(*obj.data_);
+        }
+        input_file_->load_into(*obj.data_);
```

Comment:
This **only works** for H5 input files since ROOT input file reader does not implement a copy method which moves the object data along entries.

Quick hack: Insert an extra check on which type of reader it is?